### ISSUE TITLE
Demography Workshop Datahub Request 

### ISSUE THREAD

**Issue by wrathofquan** [berkeley-dsep-infra/hubs#4413]
# Summary

Each year the Demography department hosts a [week-long workshop](https://populationsciences.berkeley.edu/population-center/annual-workshop-on-formal-demography/#:~:text=The%209th%20Annual%20Workshop%20on,by%20the%20Berkeley%20Population%20Center.) on statistical methods in June with researchers visiting from all over the world. Following up on the success we've had using the R Datahub for [C175](https://github.com/josh-goldstein-git/econ_demog_c175_2023) for instruction, we'd love to be able to run a workshop hub much like what is offered at [workshop.datahub.berkeley.edu](https://workshop.datahub.berkeley.edu/).

Let me know if this considered a duplicate of #4244 which I opened to explore more general use of datahub for demography instruction. 

# User Stories

- Workshop instructors and faculty can have all of their instructional materials in a Datahub that is consistent across users. 
- Attendees won't have to worry about managing their own compute environments. 

# Acceptance criteria

- Users can navigate to something like: demog.datahub.berkeley.edu, authenticate without calnet (and potentially manage their own credentials), access the workshop files in RStudio or native R kernel. 

# Important information

- This workshop is in June but we are actively developing the curriculum and materials now. It would be very helpful to us to have a running hub for testing as soon as is feasible - I understand work tends to be done at start of semester, so we can certainly be flexible. 
- We expect somewhere between 20-50 users
- Datasets used in the workshop will be larger than what the current workshop datahub likely can currently handle. We'd like to have a 4GB allocation of RAM per user. 
- We'd want to keep the pods alive for a few months after the workshop to give users a chance to save their work. 

# Tasks to complete

**Comment by balajialg** [comment]
Thanks, @wrathofquan for raising this issue. We will discuss this request during our sprint planning meeting to get back with a realistic estimate of when we can execute this request. 

On another note, to unblock you from testing your notebooks asap, if we increase memory for the workshop hub to 4GB RAM during the duration of this workshop, then would you be open to using that hub for this workshop? or are you specific about the demography department branding for this hub? In case if you want to test your notebooks as part of the workshop hub, then I will share the credentials for the hub.

**Comment by wrathofquan** [comment]
Awesome, thank you @balajialg. That's a great idea. Increasing resources on the current workshop hub will work too though the concerns there still might be (1) if there are other unforeseen workshops or instruction activity on that hub during the same time, what the performance implications, if any, might be? and (2) if there are any security issues around how credentials work (can DM separately about it). The branding would be ideal but not itself a blocker to delivering the workshop.

**Comment by balajialg** [comment]
@wrathofquan The only other workshop I can foresee using workshop hub during June is the [National Workshop on Data Science Education](https://data.berkeley.edu/2023workshop) which is between June 27th and 30th. Since your workshop is scheduled for the first two weeks of June, I don't see any major issues. In addition, the user count for your workshop is relatively low so I am hopeful that performance issues won't be a concern. 

Re: Security issues - The way this hub works currently is that users will be prompted to enter preconfigured password for accessing https://workshop.datahub.berkeley.edu/. Is that something you are comfortable with? Please do let us know your security concerns. I think the @shaneknapp and others in the Infra admin team can work with you to resolve your queries asap.

**Comment by balajialg** [comment]
@wrathofquan - @ericvd-ucb plans to use the workshop hub from June 20th. So, that gives us an opportunity to change the password for the workshop hub as you had requested previously. Can you pls mail us the password you want to use for the workshop hub? 

@ericvd-ucb We can recycle the password after the demog workshop if required. 

cc @ryanlovett

**Comment by balajialg** [comment]
@wrathofquan You can test the password change in https://workshop-staging.datahub.berkeley.edu/. This will get merged to prod along with similar commits asap.

**Comment by wrathofquan** [comment]
@balajialg thanks for your help with this. I believe we are all set to revert the PR around the password change as we'll be keeping the default for now.

**Comment by balajialg** [comment]
@wrathofquan Hope everything is going smooth with using Workshop hub?

**Comment by wrathofquan** [comment]
@balajialg thank you for checking in. So far everything is running smoothly!

**Comment by balajialg** [comment]
Sounds great. Thanks for the update

**Comment by balajialg** [comment]
Closing this github issue since the workshop got completed. @wrathofquan Please reopen this issue if there are any open items to be considered. Hope you had a successful workshop!
